### PR TITLE
Sets

### PR DESCRIPTION
**Pull request by svouse**
Documentation fix for project, filled in methods for ResourceSet.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-devtools/69)

### CODE REVIEW

File: src/model/resource_set.js [hunk 1]
```diff
@@ -19,8 +19,130 @@
  */
 
 /**
- * @class ResourceSet is the top level interface for sets of parts of a project.
+ * @class ResourceSet is the top level interface for sets of parts of a project,
+ *     not including the project itself. Resource sets can only contain one type
+ *     of resource, and cannot contain multiple resources with the same name.
+ *
  */
  class ResourceSet {
```
Comment (AnmAtAnm) [inline src/model/resource_set.js]: I'm guessing a ResourceSet is itself a type of Resource (it has a name, and can be loaded and saved).

File: src/model/resource_set.js [hunk 2]
```diff
@@ -19,8 +19,130 @@
...
+     * The constructor for type of resource the set manages.
+     * @type {Constructor}
+     */
+     this.resourceConstructor = resourceConstructor;
+  }
+
+  /**
+   * Writes the resource set to local storage.
+   */
+  saveSetToLocalStorage() {
```
Comment (AnmAtAnm) [inline src/model/resource_set.js]: Let's implement the local storage bits at a higher level, perhaps by passing a storage mechanism into each save method, that has the means to assign strings (file contents) into named locations (filenames, or local storage). Then the Resource subclasses won't have to worry about how they are being stored.

**Comment by rachel-fenichel** [comment]
Reviewed 1 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/model/project.js, line 25 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/69#-Knv6cZYfrZi0fTZk0Q5:-Knv6cZYfrZi0fTZk0Q6:b-v6bly3) ([raw file](https://github.com/google/blockly-devtools/blob/c527769496f016d717eac5a93b53a734509d95ff/src/model/project.js#L25)):*
> ```JavaScript
> 
> /**
>  * @class Project is a collection of one or more libraries along with
> ```

To change now or later: this paragraph can be a lot shorter.  Use active voice.  Cut filler words.  Be blunt.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/69)*

**Comment by svouse** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/model/project.js, line 25 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/69#-Knv6cZYfrZi0fTZk0Q5:-Knv8sjByC2osgiTX_4k:b5z2ux) ([raw file](https://github.com/google/blockly-devtools/blob/c527769496f016d717eac5a93b53a734509d95ff/src/model/project.js#L25)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

To change now or later: this paragraph can be a lot shorter.  Use active voice.  Cut filler words.  Be blunt.
</blockquote></details>

Agreed; cut down parts that were less useful

---

*[src/model/resource_set.js, line 27 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/69#-Knv3bRVijJZo-6go4mp-r1-27:-Knv8lBO2YVI7_xKLK-s:baof6d3) ([raw file](https://github.com/google/blockly-devtools/blob/c527769496f016d717eac5a93b53a734509d95ff/src/model/resource_set.js#L27)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

I'm guessing a ResourceSet is itself a type of Resource (it has a name, and can be loaded and saved).
</blockquote></details>

True; I'll update it to extend Resource

---

*[src/model/resource_set.js, line 57 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/69#-Knv3bRVijJZo-6go4mp-r1-57:-Knv8nYFmVylCDiZV9RA:b-202nc7) ([raw file](https://github.com/google/blockly-devtools/blob/c527769496f016d717eac5a93b53a734509d95ff/src/model/resource_set.js#L57)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Let's implement the local storage bits at a higher level, perhaps by passing a storage mechanism into each save method, that has the means to assign strings (file contents) into named locations (filenames, or local storage). Then the Resource subclasses won't have to worry about how they are being stored.
</blockquote></details>

I dig it; added a TODO in resource.js (also in this method now)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/69)*

**Comment by AnmAtAnm** [comment]
I expect the dirty/save/load methods will evolve, but this is good for now.

LGTM.

**Comment by rachel-fenichel** [comment]
lgtm, but wait for Andrew as well

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/69#-:-KnvAvSRubJ-yQ3KM2jY:boy7udb)*

**Pull request merged**